src/core/common/message.cpp

```diff
@@ -98,7 +98,7 @@ ThreadError Message::Init(void)
     return kThreadError_None;
 }
 
-Message *Message::New(uint8_t aType, uint16_t aReserved)
+Message *Message::New(uint8_t aType, uint16_t aReserved, bool aLinkSecurityEnabled)
```

Comment:
I think that would be preferable. That way the user always defaults to sending secure packets and must opt-in to sending insecure packets.